libcudacxx/include/cuda/std/__cccl/architecture.h

```diff
@@ -36,4 +39,49 @@
 
 #define _CCCL_ARCH(...) _CCCL_ARCH_##__VA_ARGS__##_()
 
+// Determine the endianness
+
+#define _CCCL_ENDIAN_LITTLE() 0xDEAD
+#define _CCCL_ENDIAN_BIG()    0xFACE
+#define _CCCL_ENDIAN_PDP()    0xBEEF
+
+#if _CCCL_COMPILER(NVRTC) || (_CCCL_COMPILER(MSVC) && (_CCCL_ARCH(X86_64) || _CCCL_ARCH(ARM64)))
+#  define _CCCL_ENDIAN_NATIVE() _CCCL_ENDIAN_LITTLE()
+#endif // _CCCL_COMPILER(NVRTC)
+
+#if !defined(_CCCL_ENDIAN_NATIVE) && __LITTLE_ENDIAN__
```

Comment:
Should those rather be 
```suggestion
#elif __LITTLE_ENDIAN__
```

Reply:
Applies throughout

Reply:
Somehow I think this approach is more clear.. But I can rework it if you want